libpromises/generic_agent.c

```diff
@@ -1636,6 +1630,30 @@ void GenericAgentConfigApply(EvalContext *ctx, const GenericAgentConfig *config)
     }
 }
 
+bool CheckAndGenerateFailsafe(const char *inputdir, const char *input_file)
+{
+    char failsafe_path[CF_BUFSIZE];
+    
+    if (strlen(inputdir) + strlen(input_file) > sizeof(failsafe_path) - 2)
+    {
+        Log(LOG_LEVEL_ERR,
+            "Unable to generate path for %s/%s file. Path too long.",
+            inputdir, input_file);
+        /* We could create dynamically allocated buffer able to hold the 
+           whole content of the path but this should be unlikely that we
+           will end up here. */
```

Comment:
Just FYI, what you mention in your comment can easily be done using `xasprintf()`, no need for error checking then.